### PR TITLE
Baby-stepping show millimeters on LCD

### PR DESCRIPTION
**Pull request by edwilliams16**
Change babystepping so that mm not steps are displayed on the LCD.

### CODE REVIEW

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -576,17 +576,20 @@ static void lcd_status_screen() {
...
         lcdDrawUpdate = LCDVIEW_REDRAW_NOW;
-        thermalManager.babystep_axis(axis, distance);
-        babysteps_done += distance;
+        thermalManager.babystep_axis(axis, babystep_increment);
+        babysteps_done += babystep_increment;
+        distance_done += distance_increment;
       }
-      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, itostr3sign(babysteps_done));
+      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, ftostr43sign(distance_done));
       if (LCD_CLICKED) lcd_goto_previous_menu(true);
```
Comment (Blue-Marlin) [inline Marlin/ultralcd.cpp]: If you don't need `babysteps_done` any more - drop it everywhere.

Comment (edwilliams16) [inline Marlin/ultralcd.cpp]: I kept it in the latest update to get rid of some irritating rounding error from incrementing a float.

File: Marlin/ultralcd.cpp [hunk 2]
```diff
@@ -575,29 +575,31 @@ static void lcd_status_screen() {
...
         encoderPosition = 0;
         lcdDrawUpdate = LCDVIEW_REDRAW_NOW;
-        thermalManager.babystep_axis(axis, distance);
-        babysteps_done += distance;
+        thermalManager.babystep_axis(axis, babystep_increment);
+        babysteps_done += babystep_increment;
+        distance_done = (float) ( (1000 * babysteps_done) / planner.axis_steps_per_mm[axis] ) / 1000.0f;// 1000 to print 3 Dec Places
       }
-      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, itostr3sign(babysteps_done));
+      if (lcdDrawUpdate) lcd_implementation_drawedit(msg, ftostr43sign(distance_done));
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp]: If you do this here instead…

``` cpp
if (lcdDrawUpdate)
  lcd_implementation_drawedit(msg, ftostr43sign(
    0.001f * ((1000 * babysteps_done) / planner.axis_steps_per_mm[axis])
  ));
```

…then you don't need to define `distance_done` above, or set it to `0.0` below. And this won't be called except when the value changes, so it doesn't cost any extra CPU.

**Comment by thinkyhead** [comment]
To squash commits is pretty straightforward from the command-line. `git rebase -i` opens up your default editor and you can use "fixup" to combine commits together. When done you can use `git push -f` to push the new commit(s).

File: Marlin/ultralcd.cpp (hunk 2 above)
Comment (edwilliams16) [inline Marlin/ultralcd.cpp]: That is certainly simpler than carrying distance_done around all over the code.  
   Right now, I'm working directly with my cloned repository at github.  Once I've figured out how to connect that up with my Mac, I'll have a command-line (and a decent editor) to work with.

**Comment by edwilliams16** [comment]
OK.  I thing I got the commits squashed together on my end - but it looks chaotic over here for such a simple fix...
  I usually work in emacs shell-mode and I haven't yet made that git friendly.
  At least it works fine on my machine...

**Comment by thinkyhead** [comment]
If you have squashed your commits, please use `git push -f` to push the fixed-up commit(s). Otherwise, I will need to clone your branch, do the squashes myself, and create a new PR.

**Comment by edwilliams16** [comment]
Do I have access to the command line on my github account?  That's where I've been making the pull requests via the web.  Or do I need to bypass that and somehow push to the Marlin account from my Mac?
    Hopefully after I get beyond being a total git novice, I'll be less trouble.

**Comment by lrpirlet** [comment]
@edwilliams16 ... I know the feeling...

I am not a mac owner, but this URL does shows all the bits and pieces I
have appreciated...
https://git-scm.com/download/mac is the page with ref to a mac tool (seems
to be a command line interface as I like it),
There is the, or should is say THE reference book: _Pro Git book
https://git-scm.com/book_

Then you'll get help from the community to fill the gap...(gaps :-) )

2016-06-15 6:48 GMT+02:00 edwilliams16 notifications@github.com:

**Comment by thinkyhead** [comment]
> after I get beyond being a total git novice

I added some scripts to help working with Marlin and git from the command-line. They live in `Marlin/buildroot/share/git` and are described at #3567. The one script it doesn't include is one I would name `mfinit` which would just do this to make sure your repo is ready for the other scripts:

```
git remote add upstream git@github.com:MarlinFirmware/Marlin.git
```

To master git you should go through this interactive tutorial: http://learngitbranching.js.org/

**Comment by thinkyhead** [comment]
Here is how I'm cleaning up this PR…

```
$ git remote add edwilliams16 git@github.com:edwilliams16/Marlin.git
$ git fetch edwilliams16
$ git checkout edwilliams16/RCBugFix -b rc_babysteps_mm
$ mfrb
```

Opens up an editor, where I edit it to say:

```
reword 780725f Update ultralcd.cpp
f 44d1903 Update ultralcd.cpp   
f 5fd480c Update ultralcd.cpp   
f 4c141b2 Update ultralcd.cpp   
f ff7d869 Update ultralcd.cpp   
f 9ff3bde put back file   
```

I save this, the editor reopens so the "reword" commit can be edited.
After saving, the `git rebase -i` is done, so I create a new PR…

```
$ mfpr
```

And the browser opens up, ready for me to create a PR.

**Comment by CONSULitAS** [comment]
@edwilliams16 
I have a Mac, too.

It is quite easy:
- Download GitHub Desktop: https://desktop.github.com
- Install
- You will have the command line tools and you can use them in the Terminal app.
- For most tasks you can use the GUI app.

**Pull request closed without merging**